python/tests/complex_type_tests/test_arr_of_arr.py

```diff
@@ -0,0 +1,152 @@
+                      c4_arr VARCHAR ARRAY ARRAY NOT NULL)"""
+        self.data = [
+            {
+                "id": 0,
+                "c1_arr": [[1, 2, None], [None]],
+                "c2_arr": [[1, 2]],
+                "c3_arr": [["bye", None], ["ciao"], [None]],
+                "c4_arr": [["hi", "hello"]],
+            },
+            {
+                "id": 1,
+                "c1_arr": [[4, 5], [8]],
+                "c2_arr": [[3]],
+                "c3_arr": [["adios", "farewell", None], ["sayonara"], [None]],
+                "c4_arr": [["ola"]],
+            },
+        ]
+
+
+class cmpxtst_arr_of_arr_unnest_int(TstView):
+    def __init__(self):
+        self.data = [
+            {"id": 0, "c1_val": [1, 2, None], "idx": 1},
+            {"id": 0, "c1_val": [None], "idx": 2},
+            {"id": 1, "c1_val": [4, 5], "idx": 1},
+            {"id": 1, "c1_val": [8], "idx": 2},
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW arr_of_arr_unnest_int AS SELECT
+                      id,  c1_val, idx
+                      FROM arr_of_arr_tbl,
+                      UNNEST(c1_arr) WITH ORDINALITY AS t (c1_val, idx)"""
+
+
+class cmpxtst_arr_of_arr_unnest_element_int(TstView):
+    def __init__(self):
+        self.data = [
+            {"id": 0, "c11_elmnt": 1, "idx": 1},
+            {"id": 0, "c11_elmnt": 2, "idx": 2},
+            {"id": 0, "c11_elmnt": None, "idx": 3},
+            {"id": 1, "c11_elmnt": 4, "idx": 1},
+            {"id": 1, "c11_elmnt": 5, "idx": 2},
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW arr_of_arr_unnest_element_int AS SELECT
+                      id,  c11_elmnt, idx
+                      FROM arr_of_arr_tbl,
+                      UNNEST(c1_arr[1]) WITH ORDINALITY AS t (c11_elmnt, idx)"""
+
+
+class cmpxtst_arr_of_arr_unnest_varchar(TstView):
+    def __init__(self):
+        self.data = [
+            {"id": 0, "c3_val": ["bye", None], "idx": 1},
+            {"id": 0, "c3_val": ["ciao"], "idx": 2},
+            {"id": 0, "c3_val": [None], "idx": 3},
+            {"id": 1, "c3_val": ["adios", "farewell", None], "idx": 1},
+            {"id": 1, "c3_val": ["sayonara"], "idx": 2},
+            {"id": 1, "c3_val": [None], "idx": 3},
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW arr_of_arr_unnest_varchar AS SELECT
+                      id,  c3_val, idx
+                      FROM arr_of_arr_tbl,
+                      UNNEST(c3_arr) WITH ORDINALITY AS t (c3_val, idx)"""
+
+
+class cmpxtst_arr_of_arr_unnest_element_varchar(TstView):
+    def __init__(self):
+        self.data = [
+            {"id": 0, "c31_elmnt": "bye", "idx": 1},
+            {"id": 0, "c31_elmnt": None, "idx": 2},
+            {"id": 1, "c31_elmnt": "adios", "idx": 1},
+            {"id": 1, "c31_elmnt": "farewell", "idx": 2},
+            {"id": 1, "c31_elmnt": None, "idx": 3},
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW arr_of_arr_unnest_element_varchar AS SELECT
```

Comment:
how about a test that uses an index that is out of bounds in some of the arrays?